synapseclient/table.py

```diff
@@ -1776,13 +1776,14 @@ def from_data_frame(cls, schema, df, filepath=None, etag=None, quoteCharacter='"
 
             test_import_pandas()
             import pandas as pd
-            for col in schema.columns_to_store:
-                if col['columnType'] == 'DATE':
-                    def _trailing_date_time_millisecond(t):
-                        if isinstance(t, str):
-                            return t[:-3]
-                    df[col.name] = pd.to_datetime(df[col.name], errors='coerce').dt.strftime('%s%f')
-                    df[col.name] = df[col.name].apply(lambda x: _trailing_date_time_millisecond(x))
+            if isinstance(schema, Schema):
+                for col in schema.columns_to_store:
+                    if col['columnType'] == 'DATE':
+                        def _trailing_date_time_millisecond(t):
+                            if isinstance(t, str):
+                                return t[:-3]
```

Comment:
Is this trimming the milliseconds?

Reply:
This is kind of a hacky function.  By default the units datetime unit returned is in nano seconds, so its basically a divide by 1000.